### PR TITLE
2320 generic survey case event handling

### PR DESCRIPTION
**Pull request by LukeLoze1**
Added receipting, refusals, invalid address, survey launched and tests

Run mvn clean install

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,4 +3,105 @@ Social/Strategic Survey Data Collection Response Management (RM) Case Processor
...
+Queue: case.refusals
+Example Msg:
+{
+  "event": {
+    "type": "RESPONSE_RECEIVED",
+    "source": null,
+    "channel": null,
+    "dateTime": null,
+    "transactionId": null
+  },
```
Comment (ryangrundy7) [inline README.md:60]: Wrong event

File: src/main/java/uk/gov/ons/ssdc/caseprocessor/model/dto/EventTypeDTO.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  UNDELIVERED_MAIL_REPORTED,
+  CCS_ADDRESS_LISTED,
+  FULFILMENT_CONFIRMED,
+  FIELD_CASE_UPDATED,
+  NEW_ADDRESS_ENHANCED,
+  DEACTIVATE_UAC,
+  RM_CASE_UPDATED,
+  RM_UNINVALIDATE_ADDRESS,
+  CLERICAL_ADDRESS_RESOLUTION,
+  SELECTED_FOR_NON_COMPLIANCE
```
Comment (AdamHawtin) [inline src/main/java/uk/gov/ons/ssdc/caseprocessor/model/dto/EventTypeDTO.java:30]: Best cut this down to just the events we're implementing

File: README.md [hunk 3]
```diff
@@ -31,6 +34,98 @@ The case processor listens to a Rabbit queue called `case.sample.inbound`. The m
 }
 ```
 
+#Invalid Address
```
Comment (AdamHawtin) [inline README.md:37]: Formatting comment, all these sub headings should be `### Subheading` to sit in the events section
e.g. 
```suggestion
### Invalid Address
```

File: README.md [hunk 4]
```diff
@@ -31,6 +34,98 @@ The case processor listens to a Rabbit queue called `case.sample.inbound`. The m
...
+    "invalidAddress": {
+      "reason": "Not found",
+      "notes": "Looked hard",
+      "caseId": "3b768940-6ef2-460e-bb75-e51d3a65ada4"
+    }
+  }
+}
+```
+
+###Receipts/Responses
```
Comment (AdamHawtin) [inline README.md:59]: Formatting needs a space after the hashes

File: src/main/java/uk/gov/ons/ssdc/caseprocessor/schedule/ClusterLeaderManager.java [hunk 5]
```diff
@@ -56,7 +56,7 @@ public boolean isThisHostClusterLeader() {
     ClusterLeader clusterLeader = lockedClusterLeaderOpt.get();
 
     if (clusterLeader.getHostName().equals(hostName)) {
-      log.with("hostName", hostName).info("This host is leader");
+      //      log.with("hostName", hostName).info("This host is leader");
```
Comment (AdamHawtin) [inline src/main/java/uk/gov/ons/ssdc/caseprocessor/schedule/ClusterLeaderManager.java:59]: Code left commented out. Perhaps some of these should be made debug level to reduce noise if they needed to be commented out to work on it?

File: src/main/java/uk/gov/ons/ssdc/caseprocessor/service/UacService.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+import uk.gov.ons.ssdc.caseprocessor.model.dto.UacDTO;
+import uk.gov.ons.ssdc.caseprocessor.model.entity.Case;
+import uk.gov.ons.ssdc.caseprocessor.model.entity.UacQidLink;
+import uk.gov.ons.ssdc.caseprocessor.model.repository.UacQidLinkRepository;
+import uk.gov.ons.ssdc.caseprocessor.utils.EventHelper;
+
+@Service
+public class UacService {
+  private static final String UAC_UPDATE_ROUTING_KEY = "event.uac.update";
+  private static final int CCS_INTERVIEWER_HOUSEHOLD_QUESTIONNAIRE_FOR_ENGLAND_AND_WALES = 71;
```
Comment (AdamHawtin) [inline src/main/java/uk/gov/ons/ssdc/caseprocessor/service/UacService.java:20]: CCS code

File: README.md [hunk 7]
```diff
@@ -31,6 +34,98 @@ The case processor listens to a Rabbit queue called `case.sample.inbound`. The m
 }
 ```
 
+### Invalid Address
+Queue Name: case.invalidaddress
```
Comment (ryangrundy7) [inline README.md:38]: Don't think this queue exists in docker dev?

**Review by ryangrundy7 — APPROVED** [review]

**Pull request merged**